rest-service/manager_rest/plugin_update/manager.py

```diff
@@ -0,0 +1,265 @@
+        plugins_update.deployments_to_update = deployments_to_update
+        self.sm.update(plugins_update)
+
+        temp_blueprint = self._create_temp_blueprint_from(blueprint, temp_plan)
+        plugins_update.temp_blueprint = temp_blueprint
+        plugins_update.state = STATES.UPDATING
+        self.sm.update(plugins_update)
+
+        plugins_update.execution = get_resource_manager(
+            self.sm).update_plugins(plugins_update)
+        plugins_update.state = STATES.EXECUTING_WORKFLOW
+        return self.sm.update(plugins_update)
+
+    def finalize(self, plugin_update_id=id):
+        """Executes the following procedure:
+        * Updates the original blueprint plan
+        * Changes all the deployments' blueprint back from the temp blueprint
+           to the original one
+        * Deletes the temporary blueprint
+        * Updates the plugins update state
+        """
+        plugin_update = self.get_plugins_update(plugin_update_id)
+
+        self._validate_plugins_update_state(plugin_update)
+        self._validate_execution_status(plugin_update)
+
+        plugin_update.state = STATES.FINALIZING
+        self.sm.update(plugin_update)
+
+        plugin_update.blueprint.plan = plugin_update.temp_blueprint.plan
+        self.sm.update(plugin_update.blueprint)
+
+        updated_deployments = self._get_deployments_to_update(
+            plugin_update.temp_blueprint_id)
+
+        for dep in updated_deployments:
+            dep.blueprint = plugin_update.blueprint
+            dep.updated_at = utils.get_formatted_timestamp()
+            self.sm.update(dep)
+
+        self.sm.delete(plugin_update.temp_blueprint)
+
+        plugin_update.state = STATES.SUCCESSFUL
+        self.sm.update(plugin_update)
+        return plugin_update
+
+    @staticmethod
+    def _validate_plugins_update_state(plugin_update):
+        if plugin_update.state != STATES.EXECUTING_WORKFLOW:
+            raise IllegalActionError(
+                "Cannot finalize plugins update {0}, "
+                "it's not in the {1} state.".format(plugin_update.id,
+                                                    STATES.EXECUTING_WORKFLOW))
+
+    def _validate_execution_status(self, plugin_update):
+        execution = self.sm.get(models.Execution, plugin_update.execution_id)
+        if execution.status in ExecutionState.END_STATES \
+                and execution.status != ExecutionState.TERMINATED:
+            plugin_update.state = STATES.FAILED
+            self.sm.update(plugin_update)
+            exec_stat_txt = 'failed' if \
+                execution.status == ExecutionState.FAILED else 'cancelled'
+            raise PluginsUpdateError(
+                'The execution of plugins update {0} {1}.'.format(
+                    plugin_update.id,
+                    exec_stat_txt))
+
+    def _get_deployments_to_update(self, blueprint_id):
+        return self.sm.list(models.Deployment,
```

Comment:
Not necessary to put in new func

Reply:
It has special filter and sort keys, this might introduce errors otherwise. That's why I think it's best to keep it in a func.